Modix.Services.Test/MediatrIntegrationTests.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Threading.Tasks;
+using Discord;
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+using Modix.Services.Core;
+using Modix.Services.Messages.Discord;
+using Modix.Services.Moderation;
+using Moq;
+using Moq.AutoMock;
+using NUnit.Framework;
+
+namespace Modix.Services.Test
+{
+    public class MediatrIntegrationTests
```

Comment:
On a more fundamental note, why is this an Integration Test? I get why you named it that, because you are testing across 2 concrete classes, `DiscordAdapter` and `ModerationInvitePurgingHandler`, but..... why are you doing that? Instead of asserting that `IModerationService` received a call, you could much more easily just mock an `INotificationHandler<ChatMessageReceived>` and assert directly on that. Effectively turning this into a unit test of `DiscordAdapter`.

Reply:
Yup you're right. The goal was to test that mediatr was wired up correctly in the DI container. I had a lot of issues with it causing it to not call the handlers. I'll improve the documentation.

Reply:
So, something doesn't work right when you try to just use handlers? Or this is just the result of you screwing around and experimenting.